kubesae/providers/gcp.py

```diff
@@ -46,7 +47,66 @@ def configure_gcp_kubeconfig(c, cluster=None, region=None):
     c.run(f"gcloud config set project {c.config.app}")
     c.run(f"gcloud container clusters get-credentials --region={region} {cluster}")
 
+@invoke.task(name="sync_media")
+def sync_media_tree(
+    c,
+    target_env="staging",
+    media_bucket="MEDIA_STORAGE_BUCKET_NAME",
+    local_target=None,
+    dry_run=False,
+    delete=False,
+):
+    """Sync a gcloud media tree for a given environment/namespace to another. 
+
+    Args:
+        target_env   (string, required): A deployment host defined in ansible host_vars (e.g. "production", "staging", "dev"). DEFAULT: staging
+        media_bucket (string, required): The variable name for media defined in settings and host_vars. DEFAULT: MEDIA_STORAGE_BUCKET_NAME
+        local_target (string, optional): Sets a target directory for local syncs
+        dry_run      (boolean, optional): Outputs the result to stdout without applying the action
+        delete       (boolean, optional): If set, deletes files on the target that do not exist on the source.
+
+    Usage:
+        inv production gcp.sync-media --dry-run: 
+            Will simulate a sync from production to staging using the bucket defined in MEDIA_STORAGE_BUCKET.
+        
+        inv production gcp.sync-media --dry-run --delete
+            Will display the files that will be deleted from the staging bucket defined in MEDIA_STORAGE_BUCKET.
+        
+        inv production gcp.sync-media --media-bucket="MEDIA" --delete
+            Will sync files from the bucket defined in the environment variable "MEDIA" to a staging bucket and
+            will delete objects on the staging bucket that do not exist on the production bucket.
+        
+        inv production gcp.sync-media --local-target="./media"
+            Will sync files from the production s3 bucket to "<PROJECT_ROOT>/media"
```

Comment:
Similar to the comment on the "S3" in the README, can we make this non-AWS specific?